homeassistant/components/notify_events/notify.py

```diff
@@ -114,7 +116,12 @@ def prepare_message(self, message, data) -> Message:
 
     def send_message(self, message, **kwargs):
         """Send a message."""
+        token = self.token
         data = kwargs.get(ATTR_DATA) or {}
 
         msg = self.prepare_message(message, data)
-        msg.send(self.token)
+
+        if ATTR_TOKEN in data and data[ATTR_TOKEN] and not data[ATTR_TOKEN].isspace():
```

Comment:
This looks a bit convoluted, could you replace it with something like:
```suggestion
        if data.get[ATTR_TOKEN, ""].trim():
```